### PR TITLE
US002 view job spec

### PR DESCRIPTION
**Pull request by OdhranHague**
As a Kainos Employee

I want to find job specification information about each role
So that I know what is expected for each role

Couple of things done:

Added unit and integration tests for getAllRoles
run tests command added to workflow
Also merged from US015 to get the "getRoleByID" function for the front end to work

### CODE REVIEW

**Review by UserConor — APPROVED** [review]
Code looks good

File: src/main/java/org/kainos/ea/cli/Role.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+                String responsibilities, String sharepointLink) {
+        this.roleName = roleName;
+        this.specification = specification;
+        this.capabilityName = capabilityName;
+        this.bandName = bandName;
+        this.responsibilities = responsibilities;
+        this.sharepointLink = sharepointLink;
+    }
+
+    public Role(String roleName, String specification,
```
Comment (steveh2707) [inline src/main/java/org/kainos/ea/cli/Role.java:70]: why do we have 2 constructors?

File: src/main/java/org/kainos/ea/db/DatabaseConnector.java
Comment (steveh2707) [inline src/main/java/org/kainos/ea/db/DatabaseConnector.java:1]: NOTE - this has all been changed on US012 branch

File: src/main/java/org/kainos/ea/db/DatabaseProperties.java
Comment (steveh2707) [inline src/main/java/org/kainos/ea/db/DatabaseProperties.java:1]: NOTE - this has been deleted on US012 branch

File: src/main/java/org/kainos/ea/db/RoleDao.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                    rs.getString("Responsibilities"),
+                    rs.getString("SharepointLink")
+            );
+            roleList.add(jobRole);
+        }
+        return roleList;
+    }
+
+    public Role getRoleByID(String role) throws SQLException {
+        Connection connection = databaseConnector.getConnection();
```
Comment (steveh2707) [inline src/main/java/org/kainos/ea/db/RoleDao.java:38]: Delete this line and pass database connection in as variable in this method instead.

File: src/main/java/org/kainos/ea/db/RoleDao.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+import org.kainos.ea.cli.Role;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+
+public class RoleDao {
+    private final DatabaseConnector databaseConnector = new DatabaseConnector();
```
Comment (steveh2707) [inline src/main/java/org/kainos/ea/db/RoleDao.java:12]: delete this line, database connection is passed in as variable to each method call.

**Pull request closed without merging**